### ISSUE TITLE
Average loss nan, using two gpus

### ISSUE THREAD

**Issue by erlendhade** [AlexeyAB/darknet#1451]
Hi

I recently purchased a new GTX 1080Ti, for the purpose of training my model faster, although after installing it, I am getting only "average loss: nan" during training.

I have tried to recalculate anchors, for resolution 608*608

I am getting the same error on both two and one gpu

**Comment by AlexeyAB** [comment]
@erlendhade Hi,

* What params do you use in the Makefile?
* Try to check your dataset
* Can you show screenshot of avg loss `nan` ?

**Comment by erlendhade** [comment]
Makefile: 
`GPU=1
CUDNN=1
CUDNN_HALF=0
OPENCV=1
AVX=0
OPENMP=0
LIBSO=1`

Dataset has been checked with your yolo_mark, and is OK

output from terminal:
`40: nan, nan avg loss, 0.002000 rate, 8.482391 seconds, 5120 images
Resizing
512 x 512 
 try to allocate workspace = 8388609 * sizeof(float),  CUDA allocate done! 
 try to allocate workspace = 8388609 * sizeof(float),  CUDA allocate done! 
Loaded: 1.503419 seconds
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000000, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: nan, Class: 0.000000, Obj: 0.000000, No Obj: 0.000000, .5R: 0.000000, .75R: 0.000000,  count: 1
Region 82 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000000, .5R: -nan, .75R: -nan,  count: 0
Region 94 Avg IOU: nan, Class: 0.000000, Obj: 0.000000, No Obj: 0.000000, .5R: 0`

I have read that others have rebuilt the repo, could this work, and if so how can I doit?

One other wierd thing:
All my checkpoints are beeing stored at 104 step intervals, but I have changed nothing in the code..

**Comment by erlendhade** [comment]
Update:

I downloaded newest repo, and did make, with the makefile configuration the same as above.

For some reason I am able to train yolov3 at 416*416 without getting avg loss = nan, but still get that error when i attempt to use resolution 608*608.

Furthermore, checkpoints are now saving at 100 step intervals.

could I possibly be doing something wrong when I ttempt to train at resolution 608*608?
- I have recalculated anchors for this resolution

**Comment by AlexeyAB** [comment]
* Did you recalculate anchors by using `./darknet calc_anchors` ... ?
* Did you change anchors in each of 3 `[yolo]`-layers?
* Just try to train first 1000 iterations by using only 1 GPU (without `-gpus 0,1` flag)

**Comment by erlendhade** [comment]
1) Yes, recalculated anchors for w=608 h=608
2) Yes
3) Yes, after that I can successfully train at 608 width and height with random = 0 on two gpus, but when i try random = 1 I get the nan average loss error
It also seems that I am able to train at 416 by 416 with random = 1 og two gpus.

**Comment by AlexeyAB** [comment]
This is strange, that you can't train with `random=1 width=608 height=608`
Try to train first 5000 iterations with random=0 and then train with random=1.
